### ISSUE TITLE
JtR TS failures while reprocessing pot file

### ISSUE THREAD

**Issue by claudioandre** [magnumripper/JohnTheRipper#2731]
* Side effect of warnings in opencl_misc.h;
* Blame aa8736f1fc6184411e7d4c879f9ba1a733cde700;
* Some tests had Errors. Performed 731 tests.  36 errors reprocessing the .POT files.

```
form=bitlocker-opencl             guesses:    3 0:00:00:08 DONE  [PASSED]
.pot CHK:bitlocker-opencl         guesses:    3 -show=   3 0:00:00:06 DONE : Expected count(s) (3)  [!!!FAILED4!!!   ret_val=0]  (3 val-pwd  4 inval-pwd)
form=blockchain-opencl            guesses:    5 0:00:00:00 DONE  [PASSED]
.pot CHK:blockchain-opencl        guesses:    2 -show=   2 0:00:00:00 DONE : Expected count(s) (5)  [!!!FAILED4!!!   ret_val=0]  (2 val-pwd  4 inval-pwd)
form=md5crypt-opencl              guesses:   61 0:00:00:00 DONE  [PASSED]
.pot CHK:md5crypt-opencl          guesses:   61 -show=  61 0:00:00:00 DONE : Expected count(s) (61)  [!!!FAILED4!!!   ret_val=0]  (61 val-pwd  4 inval-pwd)```
[...]
```
- It happens only on --dev=CPU (but this can change in the future)
```
$ ./jtrts.pl -noprelims -type keyring-opencl --passthru=-dev=3 -v
-------------------------------------------------------------------------------
- JtR-TestSuite (jtrts). Version 1.13, Dec 21, 2014.  By, JimF & others
- Testing:  John the Ripper 1.8.0-jumbo-1-5852-g6a8031f+ OMP [linux-gnu 64-bit AVX-ac]
--------------------------------------------------------------------------------

John Jumbo build detected.
--config=FILE option is valid
--field-separator-char=C option is valid
--pot=NAME option is valid
--encoding=NAME option is valid
ascii.chr found, inc_ascii_72acb6963bf12701dfb06c8d1b00cad4 added as a capability
alnum.chr found, inc_alnum_ee8763c850dee8e4b88ef547a8ed39b8 added as a capability
Types to filter on:
keyring-opencl
Capabilities in this build of john:
jumbo core opencl inc config_valid field_sep_valid local_pot_valid encode_valid utf8 cp1252 cp1251 koi8r cp437 cp737 cp850 cp858 cp866 iso8859-1 iso8859-15 inc_ascii_72acb6963bf12701dfb06c8d1b00cad4 inc_alnum_ee8763c850dee8e4b88ef547a8ed39b8

form=keyring-opencl               guesses: 1020 0:00:00:20 DONE  [PASSED]
FAILED line =   		while (((uint)s.c) & 0x03 && count--)
FAILED line =   		while (((uint)s.c) & 0x03 && count--)
FAILED line =   		while (((uint)s.c) & 0x03 && count--)
FAILED line =   		while (((uint)s.c) & 0x03 && count--)
.pot CHK:keyring-opencl           guesses: 1020 -show=1020 0:00:00:14 DONE : Expected count(s) (1020)  [!!!FAILED4!!!   ret_val=0]  (1020 val-pwd  4 inval-pwd)

Some tests had Errors. Performed 1 tests.  1 errors reprocessing the .POT files
Time used was 55 seconds
[claudio@super tests]$ 
```

**Comment by magnumripper** [comment]
> Side effect of warnings in opencl_misc.h

On what platform? Please post the full warnings, I can't reproduce.

**Comment by magnumripper** [comment]
OK I'm guessing "warning: cast from pointer to integer of different size [-Wpointer-to-int-cast]". I'll fix that but this issue remains. TS should ignore build warnings.

**Comment by magnumripper** [comment]
I was right. The trigger was fixed in 23b3d069 in bleeding-jumbo. Here's an actual warning prior to that from AMD's CPU device:
```
Build log: "/home/magnum/src/john/run/kernels/opencl_misc.h", line 324: warning: 
          conversion from pointer to smaller integer
  	if (((uint)dst & 0x03) == ((uint)src & 0x03)) {
  	     ^
(...)
```

**Comment by magnumripper** [comment]
BTW this is a bug in TS, not in JtR.
